src/Examples/UnitModels/Operations/skeleton_unit_testing.ipynb

```diff
@@ -0,0 +1,420 @@
+      "    contain the following acknowledgement:\n",
+      "        HSL, a collection of Fortran codes for large-scale scientific\n",
+      "        computation. See http://www.hsl.rl.ac.uk.\n",
+      "******************************************************************************\n",
+      "\n",
+      "This is Ipopt version 3.13.2, running with linear solver ma27.\n",
+      "\n",
+      "Number of nonzeros in equality constraint Jacobian...:        4\n",
+      "Number of nonzeros in inequality constraint Jacobian.:        0\n",
+      "Number of nonzeros in Lagrangian Hessian.............:        0\n",
+      "\n",
+      "Total number of variables............................:        4\n",
+      "                     variables with only lower bounds:        0\n",
+      "                variables with lower and upper bounds:        0\n",
+      "                     variables with only upper bounds:        0\n",
+      "Total number of equality constraints.................:        4\n",
+      "Total number of inequality constraints...............:        0\n",
+      "        inequality constraints with only lower bounds:        0\n",
+      "   inequality constraints with lower and upper bounds:        0\n",
+      "        inequality constraints with only upper bounds:        0\n",
+      "\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "   0  0.0000000e+00 0.00e+00 0.00e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0\n",
+      "\n",
+      "Number of Iterations....: 0\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Dual infeasibility......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Constraint violation....:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Complementarity.........:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "Overall NLP error.......:   0.0000000000000000e+00    0.0000000000000000e+00\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 1\n",
+      "Number of objective gradient evaluations             = 1\n",
+      "Number of equality constraint evaluations            = 1\n",
+      "Number of inequality constraint evaluations          = 0\n",
+      "Number of equality constraint Jacobian evaluations   = 1\n",
+      "Number of inequality constraint Jacobian evaluations = 0\n",
+      "Number of Lagrangian Hessian evaluations             = 0\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.000\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    }
+   ],
+   "source": [
+    "m.fs.skeleton.inlet.flow_mol_comp[0, \"c1\"].fix(2*pyunits.mol/pyunits.s)\n",
```

Comment:
This should also be broken up with some description of each step.

Of especial note, you need to discuss what happens when `m.fs.skeleton.initialize()` is called. Given that all of the content of the `SkeletonModel` is user written, the pre-written `initialize` method is going to be very simplistic and users need to be aware of exactly what its limitations are and we should probably provide some pointers on what to do if this fails (no details, but point to other docs (which might not exist yet).